### PR TITLE
Document static pages

### PR DESCRIPTION
**Pull request by dwaynebailey**
This extends the documentation for staticpages to explain the slugs and the code rewriting.  Here for review to get any correction if I'm wrong.  @translate/evernote and @unho would appreciate your review as I think you're the most aware of what happens on those pages.

### CODE REVIEW

File: docs/features/staticpages.rst [hunk 1]
```diff
@@ -14,11 +14,66 @@ There are three types of static pages:
...
+Use **Admin -- Static Pages** to create and manage static pages.
 
 The static pages are by default formatted using HTML. But you can use Markdown
 or RestructuredText by setting :setting:`POOTLE_MARKUP_FILTER` correctly.
+
+
+Links in static pages
+---------------------
+
+When linking to a static page externally or in any customisations your links
```
Comment (julen) [inline docs/features/staticpages.rst]: A comma after customizations doesn't hurt, some breathe.

File: docs/features/staticpages.rst [hunk 2]
```diff
@@ -14,11 +14,66 @@ There are three types of static pages:
...
    system.  Use these pages for presenting terms of service or changes in
-   licensing terms.
+   licensing terms that user must accept before they can use or continue to use
+   Pootle.
 #. Announcements -- these appear in the sidebar and can present special
-   instructions about projects.  Use the project code to ensure that this pages
-   appears as a sidebar for that project.  If you change these pages then they
-   will be presented to users when next they visit the project.
+   instructions about projects.  If you change these pages then they will be
+   presented to users when next they visit the project.  Announcement can also
```
Comment (julen) [inline docs/features/staticpages.rst]: *presented to users on their next visit.*

also, *Announcements*

File: docs/features/staticpages.rst [hunk 3]
```diff
@@ -14,11 +14,66 @@ There are three types of static pages:
...
+   make use of link rewriting to allow URLs to vary based on the language being
+   translated.
```
Comment (julen) [inline docs/features/staticpages.rst]: Rather than *translated*, *language being browsed*.

File: docs/features/staticpages.rst [hunk 4]
```diff
@@ -14,11 +14,66 @@ There are three types of static pages:
...
 
 The static pages are by default formatted using HTML. But you can use Markdown
 or RestructuredText by setting :setting:`POOTLE_MARKUP_FILTER` correctly.
+
+
+Links in static pages
+---------------------
+
+When linking to a static page externally or in any customisations your links
+would be poiting to ``/pages/$slug``, such as ``/pages/gettting_started``.
```
Comment (julen) [inline docs/features/staticpages.rst]: *pointing*

Also, I think let's rather suggest URLs with dashes, such as `/pages/getting-started/`.

File: docs/features/staticpages.rst [hunk 5]
```diff
@@ -14,11 +14,66 @@ There are three types of static pages:
...
+
+
+Special features of announcement pages
+--------------------------------------
+
+
+Naming your slug
+^^^^^^^^^^^^^^^^
+
+When creating an announcement page use a slug ``project/$project`` so that the
```
Comment (julen) [inline docs/features/staticpages.rst]: It's `projects/$lang` (with *s*)

File: docs/features/staticpages.rst [hunk 6]
```diff
@@ -14,11 +14,66 @@ There are three types of static pages:
...
+page will be used on the ``$project`` project.
+
+Other slug names may be used:
+
+- ``project/$lang`` - for an announcement page that will appear on every single project
```
Comment (julen) [inline docs/features/staticpages.rst]: It's `$lang`, without `project/`.

File: docs/features/staticpages.rst [hunk 7]
```diff
@@ -14,11 +14,66 @@ There are three types of static pages:
...
+  enabled for the ``$lang`` languages.
+- ``$lang/$project`` - for an announcement page that is specific to the
+  ``$project`` project in the ``$lang`` language.
+
+The prefered model though is to use the ``project/$project`` convention for a
```
Comment (julen) [inline docs/features/staticpages.rst]: Again `projects/$project`.

File: docs/features/staticpages.rst [hunk 8]
```diff
@@ -14,11 +14,66 @@ There are three types of static pages:
...
+single easy to maintian page.
```
Comment (julen) [inline docs/features/staticpages.rst]: *easy-to-maintain*

File: docs/features/staticpages.rst [hunk 9]
```diff
@@ -14,11 +14,66 @@ There are three types of static pages:
...
+   provide the functionality discussed here.
+
+In many cases you have URLs in announcement pages that would be identical
+except for variations in the language code.  Examples would include links to
+team wiki pages, signoff pages, progress dashboards, live test versions, etc.
+
+Any link within your announcement page that uses a fake language code of
+``/xx/`` will be rewritten with the language code for this translation.  Thus
+if you insert a link such as ``http://example.com/signoff/xx/`` then that will
+be rewritten to ``http://example.com/signoff/af/`` for a user vieing this
```
Comment (julen) [inline docs/features/staticpages.rst]: *viewing*

File: docs/features/staticpages.rst [hunk 10]
```diff
@@ -14,11 +14,66 @@ There are three types of static pages:
...
+  ``$project`` project in the ``$lang`` language.
+
+The prefered model though is to use the ``project/$project`` convention for a
+single easy to maintian page.
+
+
+Language link rewriting in Announcement pages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. pootle/apps/pootle_misc/templatetags/cleanhtml.py:rewrite_language_links()
```
Comment (julen) [inline docs/features/staticpages.rst]: I don't believe this note is relevant for users.

File: pootle/templates/admin/staticpages/_form.html [hunk 11]
```diff
@@ -33,3 +33,11 @@
 full paths to static pages.
 {% endblocktrans %}
 </p>
+<p>
+{% blocktrans trimmed %}
+In Announcement pages use <code><strong>/xx/</strong></code> within URLs to
+allow the same page to point to varying language code. The <code>/xx/</code>
```
Comment (julen) [inline pootle/templates/admin/staticpages/_form.html]: What about *to a dynamic language code*?

**Comment by julen** [comment]
GTM after corrections.

**Comment by unho** [comment]
GTM after addressing the raised concerns.

File: docs/features/staticpages.rst (hunk 10 above)
Comment (dwaynebailey) [inline docs/features/staticpages.rst]: .. something.... is a comment, won't be rendered, simply there for doc writers.

Comment (julen) [inline docs/features/staticpages.rst]: Anyway this doesn't feel like the go-to place for such comments. Also note the actual module path is prone to change.

**Pull request merged**